File: system-base/src/main/java/com/levnovikov/system_base/lifecycle/Lifecycle.java
package com.levnovikov.system_base.lifecycle;

import io.reactivex.disposables.Disposable;

/**
 * Author: lev.novikov
 * Date: 2/1/18.
 */

public interface Lifecycle {

    void subscribeUntil(LifecycleEvent event, Disposable disposable);

    void subscribeUntilDestroy(Disposable disposable);
}

File: system-base/src/main/java/com/levnovikov/system_base/lifecycle/Lifecycle.kt
package com.levnovikov.system_base.lifecycle

import io.reactivex.disposables.Disposable

/**
 * Author: lev.novikov
 * Date: 2/1/18.
 */

interface Lifecycle {

    fun subscribeUntil(event: LifecycleEvent, disposable: Disposable)

    fun subscribeUntilDestroy(disposable: Disposable)
}
